lib/naver/searchad/api/core/base_service.rb

```diff
@@ -6,20 +14,65 @@ class BaseService
           include Logging
 
           attr_reader :url
-          attr_reader :path
+          attr_reader :base_path
+          attr_accessor :request_options
```

Comment:
I think you can reduce this accessor scope to be a reader only from what your doing with it, unless you plan to set it from outside of this object?